src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -514,6 +557,19 @@ public bool TryGetValue(TKey key, out TValue value)
             return cacheItem != null;
         }
 
+        /// Removes a particular key from the cache.
+        public bool TryRemove(TKey key)
+        {
+            if (key == null)
+                throw LogHelper.LogArgumentNullException(nameof(key));
+
+            if (!_map.TryRemove(key, out var cacheItem))
+                return false;
+
+            OnItemCompacted?.Invoke(cacheItem.Value);
```

Comment:
It is confusing to call OnItemCompacted in TryRemove when we have a OnItemRemoved.

Maybe we can do better with naming?

Reply:
Same. This makes it seem like we're doing something to the items, when it's the whole cache that's being compacted. Thinking about this more, I begin to understand it, but initially it's not super clear.

Maybe use terms like "pending removal" or "staging removal"?

Removal is also overloaded here - removal from the LRU cache or removal from _compactedItems...

I suggest add a comment to `_compactedItems` about what it's used for.